### PR TITLE
104 - implement SfProperty

### PR DESCRIPTION
**Pull request by aniamusial**
# Related issue

https://github.com/DivanteLtd/storefront-ui/issues/104

# Scope of work

Implemented SfProperty - label-and-data component for product description

# Screenshots of visual changes

Screenshot from 2019-05-28 15-18-37

# Checklist

- [x] I followed [composition rules](https://github.com/DivanteLtd/storefront-ui/blob/master/docs/component-rules.md) for my component
- [x] I tested the component in most common device sizes (can be tested in Storybook [from viewport addon in top menu](https://github.com/storybooks/storybook/tree/master/addons/viewport))

### CODE REVIEW

File: src/components/molecules/SfProperty/SfProperty.html [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<div class="sf-property">
+	<slot>
+        <span class="sf-property__property">{{ property }}:</span>
```
Comment (leomp12) [inline src/components/molecules/SfProperty/SfProperty.html]: Hey @aniamusial , wdyt about `sf-property__name` here? Then you don't repeat _property_...

Comment (leomp12) [reply src/components/molecules/SfProperty/SfProperty.html]: or `sf-property__title`

File: src/components/molecules/SfProperty/SfProperty.html [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+        <router-link class="sf-property__value" to="/">{{ value }}</router-link>
```
Comment (leomp12) [inline src/components/molecules/SfProperty/SfProperty.html]: imho, would be good to provide a named slot here (and bind the value to slot), maybe `<slot name="value" :value="value">`, wdyt?
and `router-link` it's a default for vue-router, it's case specific so I think we shoudn't use by default, my suggestion is a normal link with `href="javascript:;"`

Comment (leomp12) [reply src/components/molecules/SfProperty/SfProperty.html]: I think you may also emit an event on click, returning the property object (with name and value)

**Comment by filrak** [comment]
By default it shouldn't have link. Links are provided only via slots (and should be a separate story on storybook)

File: src/components/molecules/SfProperty/SfProperty.html (hunk 1 above)
Comment (aniamusial) [reply src/components/molecules/SfProperty/SfProperty.html]: Sure we can change that, I was just follwing the proposed API :)

**Comment by aniamusial** [comment]
I've removed the link and changed it to standard slot with value

File: src/components/molecules/SfProperty/SfProperty.scss [hunk 3]
```diff
@@ -0,0 +1,41 @@
+@import "../../../css/variables";
+
+$property-align-items: center !default;
```
Comment (filrak) [inline src/components/molecules/SfProperty/SfProperty.scss]: redundant

File: src/components/molecules/SfProperty/SfProperty.scss [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+$property-justify-content: flex-start !default;
```
Comment (filrak) [inline src/components/molecules/SfProperty/SfProperty.scss]: redundant

File: src/components/molecules/SfProperty/SfProperty.scss [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+$property-font-size: 0.8rem !default;
```
Comment (filrak) [inline src/components/molecules/SfProperty/SfProperty.scss]: redundant, default is used.

File: src/components/molecules/SfProperty/SfProperty.scss [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+$property-line-height: 1.2rem !default;
```
Comment (filrak) [inline src/components/molecules/SfProperty/SfProperty.scss]: use this or margin, i prefer margin

File: src/components/molecules/SfProperty/SfProperty.scss [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+$property-margin-y: 1rem !default;
+$property-margin-x: 0 !default;
+$property__name-font-weight: $body-font-weight-primary !default;
+$property__name-desktop-color: $c-gray-primary !default;
```
Comment (filrak) [inline src/components/molecules/SfProperty/SfProperty.scss]: why desktop - it's a hover state same for every device

File: src/components/molecules/SfProperty/SfProperty.scss [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+$property__value-color: $c-green-primary !default;
+$property__value-text-decoration: underline !default;
+$property__value-font-weight: 500 !default;
```
Comment (filrak) [inline src/components/molecules/SfProperty/SfProperty.scss]: redundant, default is used

File: src/components/molecules/SfProperty/SfProperty.scss [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+
+.sf-property {
+  display: flex;
+  justify-content: $property-justify-content;
```
Comment (filrak) [inline src/components/molecules/SfProperty/SfProperty.scss]: redundant

File: src/components/molecules/SfProperty/SfProperty.scss [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+  align-items: $property-align-items;
```
Comment (filrak) [inline src/components/molecules/SfProperty/SfProperty.scss]: redundant

File: src/components/molecules/SfProperty/SfProperty.scss [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+  font-size: $property-font-size;
```
Comment (filrak) [inline src/components/molecules/SfProperty/SfProperty.scss]: redundant

File: src/components/molecules/SfProperty/SfProperty.scss [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+  line-height: $property-line-height;
```
Comment (filrak) [inline src/components/molecules/SfProperty/SfProperty.scss]: use this or margin

File: src/components/molecules/SfProperty/SfProperty.scss [hunk 13]
```diff
@@ -0,0 +1,41 @@
...
+  margin: $property-margin-y $property-margin-x;
+
+  &__name {
+    margin-right: 0.5rem;
+
+    @media (min-width: $desktop-min) {
```
Comment (filrak) [inline src/components/molecules/SfProperty/SfProperty.scss]: why only on desktop? it's same for mobile on designs

File: src/components/molecules/SfProperty/SfProperty.scss [hunk 14]
```diff
@@ -0,0 +1,41 @@
...
+
+$property-align-items: center !default;
+$property-justify-content: flex-start !default;
+$property-font-size: 0.8rem !default;
+$property-line-height: 1.2rem !default;
+$property-margin-y: 1rem !default;
+$property-margin-x: 0 !default;
+$property__name-font-weight: $body-font-weight-primary !default;
+$property__name-desktop-color: $c-gray-primary !default;
+$property__value-color: $c-green-primary !default;
```
Comment (filrak) [inline src/components/molecules/SfProperty/SfProperty.scss]: it's a color of value link (a) not value color

File: src/components/molecules/SfProperty/SfProperty.scss [hunk 15]
```diff
@@ -0,0 +1,41 @@
...
+  &__name {
+    margin-right: 0.5rem;
+
+    @media (min-width: $desktop-min) {
+      color: $property__name-desktop-color;
+    }
+  }
+
+  &__value {
+    cursor: pointer;
```
Comment (filrak) [inline src/components/molecules/SfProperty/SfProperty.scss]: it's true only for a link, not value itself

File: src/components/molecules/SfProperty/SfProperty.scss [hunk 16]
```diff
@@ -0,0 +1,41 @@
...
+    font-weight: $property__value-font-weight;
+    &:hover,
```
Comment (filrak) [inline src/components/molecules/SfProperty/SfProperty.scss]: it's true only for a link, not value itself

File: src/components/molecules/SfProperty/SfProperty.js [hunk 17]
```diff
@@ -0,0 +1,13 @@
+export default {
+  name: "SfProperty",
+  props: {
+    name: {
+      type: String,
+      default: ""
+    },
+    value: {
+      type: [String],
```
Comment (filrak) [inline src/components/molecules/SfProperty/SfProperty.js]: why it's an array?

**Pull request merged**